### DISCUSSION TITLE
How to save a simple Date data type value (on trigger function)?

### DISCUSSION THREAD

**Discussion by ichbinedgar** [supabase/supabase#4064] [Questions] [answered]
I'm trying something like the following (it fails on the authentication trigger function):

```

begin
  insert into public."Users" (id, email , name, phone_1, phone_2, id_condo, birthday  )
  values (
  new.id, 
  new.email, 
  new.raw_user_meta_data->>'name', 
  new.raw_user_meta_data->>'phone_1', 
  new.raw_user_meta_data->>'phone_2',
  (new.raw_user_meta_data->>'id_condo'::text)::uuid,
   new.raw_user_meta_data->>'birthday'

  );
  return new;
end;
```

**Comment by ichbinedgar** [comment, marked as answer]
ok, I was able to fix it by casting the value like this:

`(new.raw_user_meta_data->>'birthday'::text)::date`

However, supabase stores the value as a timestamp ... not sure why, maybe a bug?

**Reply by ichbinedgar** [reply]
PgAdmin shows the date correctly so I guess it's a bug
<img width="265" alt="Screen Shot 2021-11-26 at 1 02 44" src="https://user-images.githubusercontent.com/5431012/143540391-bd4fc5c4-a32b-414d-a8ea-708212406713.png">

**Reply by soedirgo** [reply]
Answered here: https://github.com/supabase/supabase/issues/4076

**Reply by pedropankaj** [reply]
its working thanks

**Comment by jarvismartin** [comment]
This took a lot of time and effort to find! 
I realize this is more of a Postgres thing, not a Supabase thing, but how hard would it be to add this as an example in the docs? It doesn't seem like it would fit in the [data types documentation](https://supabase.com/docs/guides/database/tables#data-types), but it would be great to see there as a gotcha to avoid. 
It was easy to find the format, and easy to find that it has to be ISO. I figured there had to be a type problem, but where? The `<input type="date">` was giving me the right format, as a string. Was Typescript doing something weird? I used `date.toISOString()` and formatted from there. I looked SQL and Postgres DATE format. The error from the triggered function:

> AuthApiError: Database error saving new user
>     at .../node_modules/@supabase/gotrue-js/dist/main/lib/fetch.js:41:20
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
>   __isAuthError: true,
>   status: 500

isn't specific enough (for me to figure out what to do). I knew sign up failed, but I had to test step by step to find out why it failed. Maybe just posting the error here will help other people find this solution, but I'd be happy to do a PR to add this gotcha to the docs. I'm just not sure where it should go.

Thanks so much for this answer @ichbinedgar!!!
